### PR TITLE
chore(examples): fix version bumping of examples

### PR DESCRIPTION
**Pull request by flochaz**
## Description of your changes

Currently when a version are bumped, examples cdk package got his dependencies bumped as well to match latest package that are going to be released.
1. This can't be control due to this https://github.com/lerna/lerna/issues/2326 bug.
2. Bumping dependencies can make example code buggy since it won't be updated to go around potential breaking changes ...

This PR leverage [versionning life cycle](https://github.com/lerna/lerna/tree/6cb8ab2d4af7ce25c812e8fb05cd04650105705f/commands/version#lifecycle-scripts) to force versioning of deps and workaround https://github.com/lerna/lerna/issues/2326 .

NB: Examples will always be behind if we don't update manually the dependencies.

### Alternative

Remove `examples/cdk` from lerna.json and add lint, build and test specific scripts to either root `package.json` or in github actions and husky scripts ...

### How to verify this change

1. Bump version `npx lerna version --force-publish=\* --no-git-tag-version --no-push  --conventional-commits`
2. Check that examples package.json and lock for no change on deps

Build is passing https://github.com/awslabs/aws-lambda-powertools-typescript/runs/4898015010 so package-lock.json is fixed.

Next bump then run e2e tests will demonstrate the fix

### Related issues, RFCs

#494

### PR status

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/cdk/package.json
Comment (dreamorosi) [inline examples/cdk/package.json:14]: Why are we setting a specific version? And who is going to call `npm run version`?

Comment (flochaz) [reply examples/cdk/package.json]: Examples are written against a certain version of our package. By definition its a version that has already been pushed to NPM.

Comment (flochaz) [reply examples/cdk/package.json]: lerna version is calling version

**Review by dreamorosi — APPROVED** [review]

**Review by ijemmy — COMMENTED** [review]
@flochaz  

Feel really awkward about this. But I also can't think of a better fix and want to put my thoughts before approval.

## `^0.4.0` and `0.4.0`
My main concern is that we now have `^0.4.0` (with caret) and will be override with `0.4.0` only when running `lerna`. The effect are:

1. If we release `0.4.1`, this `version` will be reverted back to 0.4.0 in `lerna` run. Any patching we did won't be applied here. We need to remember to update this every time we bump a patch version.
1. Most contributors will not know that `version` target is run automatically by `lerna`. 

## Should we keeping example up-to-date through checklist?
Users who try this example with `npm install` will use the latest patched version. This is good. However, users may copy the code from example into their project with the latest version of library.  
   * If the latest is just patching (0.4.1) or  minor (0.5.0), it should be fine as it is backward compatible.
   * If the latest is major (1.0.x), the example can break.

To avoid this, we have to either add an updating example as part of Definition of Done checklist. This is manual now.
   
## The alternative (leaving it as it is) is worse option
Alternatively, we  can leave it as it is. We risk having broken examples for users.  The users may or may not report back to us. So we might have broken examples for a very long time before any detection. 

Choosing between two choice, I prefer to have this change in than leaving it as it is.

**Review by ijemmy — APPROVED** [review]
This is blocking merge. Let's go with this for now. Will discuss with the team on adding a checklist in PR so that we don't miss updating this.

**Pull request merged**
